code/datums/mutantraces.dm

```diff
@@ -261,115 +239,241 @@
-
-		var/list/L = hex_to_rgb_list(hex)
-		for (var/i in L)
-			L[i] = min(L[i], 190)
-			L[i] = max(L[i], 50)
-		if (L.len == 3)
-			return rgb(L["r"], L["g"], L["b"])
-		return rgb(22, 210, 22)
-
-	/// Backs up the character's hair colors, then does fix_colors on those colors if the FIX_COLORS flag is set
-	proc/AppearanceSetter(var/mob/living/carbon/human/Q, var/mode as text)
-		if(!ishuman(Q) || !(Q?.bioHolder?.mobAppearance))
-			return // please dont call set_mutantrace on a non-human non-appearanceholder
+				if (src.l_limb_leg_type_mutantrace)
+					if (L.limbs.l_leg || src.ignore_missing_limbs == 1)
+						var/obj/item/parts/human_parts/leg/limb = new /obj/item/parts/human_parts/leg/left(L)
+						if (istype(limb))
+							qdel(L.limbs.l_leg)
+							limb.quality = 0.5
+							L.limbs.l_leg = limb
+							limb.holder = L
+							limb.remove_stage = 0
+				//////////////HEAD//////////////////
+				L.organHolder.head.MakeMutantHead(HEAD_HUMAN)
 
-		src.AH = Q.bioHolder.mobAppearance // i mean its called appearance holder for a reason
+	proc/organ_mutator(var/mob/living/carbon/human/O, var/mode as text, var/drop_tail)
+		if(!ishuman(O) || !(O?.organHolder))
+			return // hard to mess with someone's organs if they can't have any
 
-		switch(mode)
-			if("set")	// upload everything, the appearance flags'll determine what gets used
-				AH.mob_color_flags = src.mutant_color_flags
-				// Store our old colors, just in case we stop being a weirdo
-				AH.customization_first_color_carry = AH.customization_first_color
-				AH.customization_second_color_carry = AH.customization_second_color
-				AH.customization_third_color_carry = AH.customization_third_color
-				if (src.mutant_color_flags & FIX_COLORS)
-					AH.customization_first_color = fix_colors(AH.customization_first_color)
-					AH.customization_second_color = fix_colors(AH.customization_second_color)
-					AH.customization_third_color = fix_colors(AH.customization_third_color)
+		var/datum/organHolder/OHM = O.organHolder
 
-			if("reset")
-				AH.mob_color_flags = (HAS_HAIR_COLORED_HAIR)
-				AH.customization_first_color = AH.customization_first_color_carry
-				AH.customization_second_color = AH.customization_second_color_carry
-				AH.customization_third_color = AH.customization_third_color_carry
-				detail_1 = null
-				detail_2 = null
-				detail_3 = null
-				detail_over_suit = null
+		switch(mode)
+			if("set")
+				if(!src.mutant_organs.len)
+					return // All done!
+				else
+					for(var/mutorgan in src.mutant_organs)
+						if (mutorgan == "tail") // Not everyone has a tail. So just force it in
+							if (OHM.tail)
+								var/obj/organ_drop = OHM.tail
+								OHM.drop_organ("tail")
+								if (!drop_tail)
+									qdel(organ_drop)
+						else // But everyone has everything else.
+							var/obj/item/organ/org = OHM.get_organ(mutorgan)
+							if (!istype(org) || org?.robotic) // No free organs, trade-ins only, keep ur robotic stuff
+								continue
```

Comment:
relevant to above: the `istype(org)` will skip make this skip butts, as they are not `obj/item/organ`